src/main/java/seedu/utility/FinancialTracker.java

```diff
@@ -9,108 +8,93 @@
 import java.util.ArrayList;
 
 public class FinancialTracker {
-    private ArrayList<Entry> financialEntries;
+    private ArrayList<Expense> expenses;
+    private ArrayList<Income> incomes;
 
     public FinancialTracker() {
-        this.financialEntries = new ArrayList<>();
+        this.expenses = new ArrayList<>();
+        this.incomes = new ArrayList<>();
     }
-
-    public int getSize() {
-        return financialEntries.size();
-    }
-
-    public boolean isEmpty() {
-        return financialEntries.isEmpty();
+    
+    public void addExpense(Expense expense) {
+        expenses.add(expense);
+        assert !expenses.isEmpty();
     }
 
-    public Entry addEntry(Entry entry) {
-        financialEntries.add(entry);
-        return entry;
+    public void addIncome(Income income) {
+        incomes.add(income);
+        assert !incomes.isEmpty();
     }
-
-    public Entry removeEntry(int itemIndex) {
-        Entry deletedEntry = financialEntries.get(itemIndex);
-        financialEntries.remove(itemIndex);
-        if (deletedEntry instanceof Expense) {
-            Expense.decrementSize();
-        } else if (deletedEntry instanceof Income) {
-            Income.decrementSize();
+    
+    public Expense removeExpense(int expenseIndex) throws ExpenseEntryNotFoundException {
+        try {
+            return expenses.remove(expenseIndex - 1);
```

Comment:
Maybe can have a separate function that does expenseIndex - 1 ie offsetIndexForArray() to increase readability.